File: Day9/Part1.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace AdventOfCode.Day9
{
    class Part1
    {
        private static HashSet<string> traveledTo = new HashSet<string>();

        private static Dictionary<string, int[]> moveMatrix = new Dictionary<string, int[]> {
            { "U", new int[] { -1,  0 } },
            { "D", new int[] {  1,  0 } },
            { "L", new int[] {  0, -1 } },
            { "R", new int[] {  0,  1 } },
        };

        static void Temp()
        {
            string[] lines = System.IO.File.ReadAllLines(@"C:\Users\User\Downloads\advent.txt");

            int[] head = new int[] { 4, 0 };
            int[] tail = new int[] { 4, 0 };
            foreach (string line in lines)
            {
                string[] move = line.Split(" ");
                for (int i = 0; i < Int32.Parse(move[1]); i++)
                {
                    head = head.Zip(moveMatrix[move[0]], (x, y) => x + y).ToArray();

                    double distance = Math.Sqrt(Math.Pow(head[0] - tail[0], 2) + Math.Pow(head[1] - tail[1], 2));
                    if (distance > Math.Sqrt(2))
                    {
                        tail = head.Zip(tail, (x, y) => Math.Sign(x - y) + y).ToArray();
                    }

                    traveledTo.Add(tail[0] + ":" + tail[1]);
                }
            }

            Console.WriteLine(traveledTo.Count);
        }
    }
}

File: 2022/Day2/Part1.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

public class Part1 {

    private static final int[][] pointsMatrix = { {3, 6, 0},
                                                  {0, 3, 6},
                                                  {6, 0, 3} };

    public static void main(String[] args) throws IOException {
        String file = "C:\\Users\\User\\Downloads\\advent.txt";

        int score = 0;
        try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
            for(String line; (line = reader.readLine()) != null; ) {
                String[] moves = line.split(" ");

                int opponent = moves[0].charAt(0) - 'A';
                int player = moves[1].charAt(0) - 'X';

                score += (player + 1) + pointsMatrix[opponent][player];
            }
        }

        System.out.println("Score: " + score);
    }
}
